### DISCUSSION TITLE
Custom Non-LLM evaluators/scores through UI

### DISCUSSION THREAD

**Discussion by simonwh** [langfuse/langfuse#4484] [Ideas]
### Describe the feature or potential improvement

Support for non-LLM evaluators for both traces and experiments, and to be able to define and run them in Langfuse, and not locally, would be extremely helpful. While LLM judge is great, there's a big need to use something like "ExactMatch", "ListContains", "ValidJSON" etc., even when domain experts are the prompt engineers (for example imagine categorizing conversations - you would want a combination of LLM + non-LLM evaluators to run in experiments and on your prod traces)

I would say it's one of the (few) places where braintrust.dev excels over Langfuse atm.

Being able to write them ourselves with typescript/python in the UI would be the most flexible, but it would also be very helpful with pre-made evaluators (just like you have with LLM-as-judge). 

Some inspiration:
* [Braintrust custom scores](https://www.braintrust.dev/docs/guides/functions/scorers)
* [Braintrust AutoEvals library](https://github.com/braintrustdata/autoevals)

**Comment by marcklingen** [comment]
Thanks for sharing! I agree that the ability to run custom non-LLM evaluators in Langfuse would be helpful. I appreciate you taking the time to share this. I will keep you in the loop, as this is definitely something we are considering adding to Langfuse.

**Comment by sebastian-montero** [comment]
Any updated on this?

**Comment by wukimidaire** [comment]
yeah new user of langfuse, coming from langsmith, kind of surprised LLM as a judge is the default without this simple option for exact match?

**Comment by marliessophie** [comment]
No update on this yet unfortunately, but we have it on the mid-term roadmap and will continue posting updates here. Thanks!

**Comment by aandis** [comment]
Any updates on when this will be implemented?
